sources/Algo/src/Algo.cpp

```diff
@@ -246,5 +249,277 @@ ControllerInterfaceDefinitionAlgorithm::operator()(const NodePtr& node) {
+  for (const auto& asso : singleassociations) {
+    singleAssociationsMap[asso.id] = asso;
+  }
+  std::unordered_map<std::string, AssemblingXmlDocument::MultipleAssociation> multiAssociationsMap;
+  for (const auto& asso : multiassociations) {
+    multiAssociationsMap[asso.id] = asso;
+  }
+
+  for (const auto& automaton : automatons) {
+    // Check that the automaton library is accessible
+    if (!doesLibraryExist(automaton.lib)) {
+      LOG(warn) << MESS(DynModelLibraryNotFound, automaton.lib, automaton.id) << LOG_ENDL;
+      continue;
+    }
+    automatonsById_[automaton.id] = automaton;
+
+    for (const auto& macro : automaton.macroConnects) {
+      auto singleassoc = singleAssociationsMap.find(macro.id);
+      if (singleassoc != singleAssociationsMap.end()) {
+        dispatchAutomatonSingle(automaton, macro, singleassoc->second);
+        continue;
+      }
+
+      auto multiassoc = multiAssociationsMap.find(macro.id);
+      if (multiassoc != multiAssociationsMap.end()) {
+        dispatchAutomatonMulti(automaton, macro, multiassoc->second);
+        continue;
+      }
+    }
+  }
+}
+
+bool
+DynModelAlgorithm::MacroConnect::operator==(const MacroConnect& other) const {
+  return dynModelId == other.dynModelId && macroConnectionId == other.macroConnectionId;
+}
+
+bool
+DynModelAlgorithm::MacroConnect::operator!=(const MacroConnect& other) const {
+  return !((*this) == other);
+}
+
+bool
+DynModelDefinition::MacroConnection::operator==(const MacroConnection& other) const {
+  return id == other.id && elementType == other.elementType && connectedElementId == other.connectedElementId;
+}
+
+bool
+DynModelDefinition::MacroConnection::operator!=(const MacroConnection& other) const {
+  return !((*this) == other);
+}
+
+bool
+DynModelDefinition::MacroConnection::operator<(const MacroConnection& other) const {
+  return (id + std::to_string(static_cast<unsigned int>(elementType)) + connectedElementId) <
+         (other.id + std::to_string(static_cast<unsigned int>(other.elementType)) + other.connectedElementId);
+}
+
+bool
+DynModelDefinition::MacroConnection::operator<=(const MacroConnection& other) const {
+  return (*this) < other || (*this) == other;
+}
+
+bool
+DynModelDefinition::MacroConnection::operator>(const MacroConnection& other) const {
+  return !((*this) <= other);
+}
+
+bool
+DynModelDefinition::MacroConnection::operator>=(const MacroConnection& other) const {
+  return (*this) > other || (*this) == other;
+}
+
+void
+DynModelAlgorithm::processDynModelShuntConnection(const NodePtr& node) {
+  // Connect All nodes of voltage level
+  auto vl = node->voltageLevel.lock();
+  const auto& macroConnections = macroConnectByVlForShuntsId_.at(vl->id);
+
+  for (const auto& macroConnection : macroConnections) {
+    models_.usedMacroConnections.insert(macroConnection.macroConnectionId);
+    const auto& automaton = automatonsById_.at(macroConnection.dynModelId);
```

Comment:
how are we sure that this automaton exists?

Reply:
By construction, the macro connections in the macroConnectByVlForShuntsId_ come from dynamic automaton definition, which has been dispatched

Reply:
then this should be visible in the name of the structure

Reply:
OK